Lib/http/server.py

```diff
@@ -902,6 +942,30 @@ def guess_type(self, path):
             return guess
         return 'application/octet-stream'
 
+    def parse_range(self):
+        """Return a tuple of (start, end) representing the range header in
+        the HTTP request. If the range header is missing or not resolvable,
+        None is returned. This only supports single part ranges.
+
+        """
+        range_header = self.headers.get('range')
+        if not range_header:
+            return None
+        m = re.match(RANGE_REGEX_PATTERN, range_header)
```

Comment:
```suggestion
        m = RANGE_REGEX_PATTERN.match(range_header)
```

That way, you save a call to re.compile and an isinstance check.